docs/dev/background-information/sql/migrator.mdx

```diff
@@ -0,0 +1,363 @@
+the state of the database and apply necessary schema changes. This
+design was initially implemented as a tool for TS team members to assist
+in multiversion upgrades and because it could easily be included over
+multiple deployment methods as a containerized process run separately
+from the frontend during startup ([which originally caused
+issues](https://about.sourcegraph.com/blog/introducing-migrator-service)).
+This replaced earlier go/migrate based strategies which ran in the
+frontend on startup. While the migrator can operate as a CLI tool, it's
+containerized as if it was another application which allows it to be run
+as an initContainer (in Kubernetes deployments) or as a dependent
+service (for Docker Compose deployments) to ensure that the necessary
+migrations are applied before startup of the application proper.
+Check out [RFC 469](https://docs.google.com/document/d/1_wqvNbEGhMzu45CHZNrLQYys7gl0AtVpLluGp4cbQKk/edit#heading=h.ddeuyk4t99yx) to learn more.
+
+### Important Commands
+
+The most important migrator commands are `up` and `upgrade`, with a notable mention to `drift`:
+
+-   **Up** : The default command of `migrator`, `up` ensures that
+    for a given version of the migrator, every migration defined at 
+    that build has been successfully applied in the connected database, 
+    this is specifically important to ensure *patch version migrations are run*. 
+    ***This must be run with the target version migrator image/build**.* 
+    If migrator and frontend pods are deployed in version lockstep, `up` 
+    ensures that ALL migrations required by the frontend will be successfully applied prior to boot.
+    This is a syntactic sugar over a more internal `upto` command.
+
+-   **Upgrade**: `upgrade` runs all migrations defined between two minor versions
+    of Sourcegraph and requires that other services which may access the
+    database are brought down. Before running, the database is checked
+    for and schema drifts in order to prevent a failure while attempting
+    a migration. `upgrade` relies on `stitched-migration-graph.json`.
+
+-   **Drift**: This command pulls runs a diff between the current database
+    schema and an expected definition packaged in migrator during the
+    release process. Many migrator operations run this check before 
+    proceeding to ensure the database is in the expected state.
+
+In general, the `up` command can be thought of as a **standard** upgrade
+(rolling upgrades with no downtime) while the upgrade command is what
+enables **multiversion** upgrades. In part, `up` was designed to maintain
+our previous upgrade policy and is thus run as an initContainer (or
+initContainer-like mechanism) of the frontend, i.e. between two versions
+of Sourcegraph, the antecedent sourcegraph services will continue to
+work after the consequent version's migrations have been applied.
+
+## Current Startup Dependency
+
+<div style="display: flex; flex-direction: row; align-items: flex-start;">
```

Comment:
These divs are hard to read in the md file format but its all to put an image to the side of text, and make sure its moderately responsive to browser size